### PR TITLE
Fixed behat issues

### PR DESCRIPTION
**Pull request by ClementGautier**
Select2 filters crash when we try to close them by clicking direcly on the element using Selenium Driver so we use the Javascript API to ensure that the select2 closes properly

### CODE REVIEW

**Comment by a2xchip** [comment]
Had you seen #4361 conversations about behats? We have unresolved issue with driver included in method that is currently is not resolved and i see the same in your PR. But i found way to click bypassing this selenium bug.  

Ping @pierallard

**Comment by ClementGautier** [comment]
@a2xchip Thx for the hint. I double checked and here it was the user behavior to click on the mask (Actually the user was selecting the operator first but, as the filter made a focus on the select2 when it opens you basically have to click twice in order the open the operator selector). I decided to change the execution order in a way that you first input the value and then the operator so it works as expected without magic.

File: features/Context/Page/Base/ProductEditForm.php [hunk 1]
```diff
@@ -79,8 +79,8 @@ public function findAvailableAttributeInGroup($attribute, $group)
             }
         }
 
-        // Close select2
-        $selector->click();
+        // Close select2 if open
+        $this->getDriver()->evaluateScript("jQuery('.select2-drop-mask').click();");
```
Comment (a2xchip) [inline features/Context/Page/Base/ProductEditForm.php]: evaluate vs execute as i remember from mink docs is about return value. Evaluate if you need to get js return and execute if not. Same for above...

**Comment by a2xchip** [comment]
@ClementGautier I am curious if you can try mask "workaround" in mentioned PR branch? May be it will be more elegant decision?

**Comment by ClementGautier** [comment]
@a2xchip as it is still in discussion I'll keep it the old way for now (basically we should change that at other places too).

File: features/Context/Page/Base/Grid.php [hunk 2]
```diff
@@ -196,17 +196,28 @@ public function filterBy($filterName, $value, $operator = false, DriverInterface
...
+                // that requires the user to click twice, once in order to close the select2, and then an other one to
+                // open the operator selector.
+                if (!$driver || !$driver instanceof Selenium2Driver) {
+                    throw new \InvalidArgumentException('Selenium2Driver is required to filter by a choice filter');
+                }
+
+                $driver->executeScript("jQuery('.select2-drop-mask').click();");
+            }
+
             if (false !== $operator) {
```
Comment (juliensnz) [inline features/Context/Page/Base/Grid.php]: $operator should be a string no ?

File: features/Context/Page/Base/Grid.php [hunk 3]
```diff
@@ -196,17 +196,28 @@ public function filterBy($filterName, $value, $operator = false, DriverInterface
...
 
+            if (in_array($value, ['empty', 'is empty'])) {
+                $operator = 'empty';
+            }
+
+            if (null !== $select2) {
+                // Opening a filter makes focus on the field and, in case of select2 multiselect can result in a mask
+                // that requires the user to click twice, once in order to close the select2, and then an other one to
+                // open the operator selector.
+                if (!$driver || !$driver instanceof Selenium2Driver) {
```
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: Useless check imho

File: features/Context/Page/Base/Grid.php [hunk 4]
```diff
@@ -196,17 +196,28 @@ public function filterBy($filterName, $value, $operator = false, DriverInterface
             $results = $this->getElement('Select2 results');
             $select2 = $filter->find('css', '.select2-input');
 
+            if (in_array($value, ['empty', 'is empty'])) {
+                $operator = 'empty';
+            }
+
+            if (null !== $select2) {
```
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: Remove this and spin at line 197 please

File: features/Context/Page/Base/Grid.php (hunk 2 above)
Comment (phaseinducer) [inline features/Context/Page/Base/Grid.php]: Please change the default value to `null` if possible.

**Comment by phaseinducer** [comment]
GTM

**Comment by pierallard** [comment]
GTM

**Pull request merged**